src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+abstract class Figure implements IFigure {
```

Comment:
The class name 'Figure' should not be a verb. Consider renaming it to something more indicative of what it represents, like 'AbstractFigure' or 'BaseFigure', if it's meant to provide a base for other figures.